src/fsharp/LexFilter.fs

```diff
@@ -578,10 +571,14 @@ type LexFilterImpl (lightSyntaxStatus:LightSyntaxStatus, compilingFsLib, lexer,
     let popNextTokenTup() = 
         if delayedStack.Count > 0 then 
             let tokenTup = delayedStack.Pop()
+            #if DEBUG
             if debug then dprintf "popNextTokenTup: delayed token, tokenStartPos = %a\n" outputPos (startPosOfTokenTup tokenTup)
```

Comment:
i know the `if debug then` is redundant with these debug directives but the whole point is that on release builds, it will not even include these wasteful expressions, the paths are quite hot in the lex stage so it is repeatedly doing this check & branch hundreds/thousands of times for no reason so I cannot see why you would leave it in release builds if we can remove with directives … it's messy due to the fact we have hacky test debugging code live in a production release build, this would never get out the gate on a release build of other production apps so cannot see why we accept here? we can keep in, with directives to allow the future debug testing of changes to the lexer but to include all these if checks in release builds is bizarre to me, especially in something as performance critical as compiler?